pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -253,9 +253,12 @@ func (rs *ReceiverService) DeleteReceiver(ctx context.Context, uid string, calle
 	}
 
 	// Check optimistic concurrency.
-	err = rs.checkOptimisticConcurrency(existing, version)
-	if err != nil {
-		return err
+	// Optimistic concurrency is optional for delete operations, but we still check it if a version is provided.
+	if version != "" {
+		err = rs.checkOptimisticConcurrency(existing, version)
+		if err != nil {
+			return err
+		}
```

Comment:
It would be great if we could do something like this, and log the situation when optimistic concurrency was skipped

https://github.com/grafana/grafana/blob/135f6571a9b8a5c471282b575e5fa5f670c3622e/pkg/services/ngalert/provisioning/mute_timings.go#L391-L404